velox/exec/HashProbe.h

```diff
@@ -379,25 +379,15 @@ class HashProbe : public Operator {
     // side rows with multiple matches on the build side are next to each other.
     template <typename TOnMiss>
     void advance(vector_size_t row, bool passed, TOnMiss onMiss) {
-      if (currentRow != row) {
-        if (currentRow != -1 && !currentRowPassed) {
-          onMiss(currentRow);
-        }
-        currentRow = row;
-        currentRowPassed = false;
-      }
-      if (passed) {
-        currentRowPassed = true;
+      currentRow = row;
+      currentRowPassed = passed;
+      if (currentRow != -1 && !currentRowPassed) {
```

Comment:
When will `currentRow != -1` be true, here you always initialize before the check?

With new change, we don't need currentRow or currentRowPassed at all.

Reply:
I have changed the `noMatchDetector_.finish();` as well. So when a batch is completely processed, this will be called and currentRow will be set back to -1 and currentRowPassed to false.

We don't need the currentRow or currentRowPassed as it denotes that it is kept for tracking the probe side. The logic change here is if we take the first row from the input batch, then we use the passed value of it and add to the corresponding table with/without NULLs. The currentRow and currentRowPassed will hold the value until next input row arrives.

Let me know if this description satisfies you or we can get over call and discuss this.

Reply:
CircleCI builds have failed. 
```
274: [  FAILED  ] 12 tests, listed below:
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.nullAwareAntiJoinWithFilterAndNullKey/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.nullAwareAntiJoinWithFilterAndNullKey/1, where GetParam() = 4-byte object <03-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.nullAwareAntiJoinWithFilterOnNullableColumn/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.nullAwareAntiJoinWithFilterOnNullableColumn/1, where GetParam() = 4-byte object <03-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.antiJoin/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.antiJoin/1, where GetParam() = 4-byte object <03-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.leftJoinWithFilter/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.leftJoinWithFilter/1, where GetParam() = 4-byte object <03-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.leftJoinWithNullableFilter/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.leftJoinWithNullableFilter/1, where GetParam() = 4-byte object <03-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.fullJoinWithFilters/0, where GetParam() = 4-byte object <01-00 00-00>
274: [  FAILED  ] HashJoinTest/MultiThreadedHashJoinTest.fullJoinWithFilters/1, where GetParam() = 4-byte object <03-00 00-00>
274: 
274: 12 FAILED TESTS
274:   YOU HAVE 1 DISABLED TEST
274: 
```